### ISSUE TITLE
Support for enum-casts

### ISSUE THREAD

**Issue by eschricker** [barryvdh/laravel-ide-helper#1354]
### Summary

I have defined a enum which is used in the casts of a eloquent model. On the database the field is a smallint. The generated doc-block marks the element as int. It would be great if the enum could be referenced here. 

```php
/**
 * App\Models\User.
 ...
 * @property int $role
*/
class User extends Authenticatable {
...
  protected $casts = [
    'id' => 'integer',
    'role' => UserRoles::class,
    'enabled' => 'boolean',
  ];
...
}
```

**Comment by Bosphoramus** [comment]
Nevermind, it works perfectly fine, declared the `isAdmin()` function as protected by mistake, making it public fixes it.

So this issue is already fixed, isnt it?

**Comment by Tofandel** [comment]
Yes and no, there is an issue with it, if running `php artisan ide-helper:models -M` in the `_ide_helper_models.php` file, then the result is

```
 * @property OrderStatus $status
```

But there is no `use App\Enums\OrderStatus` in this file, so the declaration is missing a full namespace (there is also the same issue with some scope return types like Builder)

**Comment by Tofandel** [comment]
Yes, this issue can be closed, the feature exists
